### PR TITLE
Add order model

### PR DESCRIPTION
**Pull request by TetianaDenchyk**

### CODE REVIEW

File: src/main/java/com/bookshop/model/Book.java [hunk 1]
```diff
@@ -61,6 +61,11 @@ public class Book {
             cascade = CascadeType.REMOVE)
     private List<CartItem> cartItems = new ArrayList<>();
 
+    @OneToMany(mappedBy = "book",
+            cascade = CascadeType.REMOVE,
+            orphanRemoval = true)
+    private Set<OrderItem> orderItems;
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/model/Book.java]: Do you really need this field here?

File: src/main/java/com/bookshop/model/User.java [hunk 2]
```diff
@@ -60,6 +61,11 @@ public class User implements UserDetails {
             cascade = CascadeType.ALL)
     private ShoppingCart shoppingCart;
 
+    @OneToMany(mappedBy = "user",
+            cascade = CascadeType.REMOVE,
+            orphanRemoval = true)
+    private Set<Order> orders;
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/model/User.java]: the same

File: src/main/java/com/bookshop/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            orderItem.setOrder(order);
+            orderItems.add(orderItem);
+            cartItem.setShoppingCart(null);
+        }
+        BigDecimal total = order.getOrderItems().stream()
+                .map(i -> i.getPrice().multiply(BigDecimal.valueOf(i.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        order.setTotal(total);
+        shoppingCart.getCartItems().clear();
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/OrderServiceImpl.java]: Let’s  divide this logic by some private methods for increasing readability

File: src/main/java/com/bookshop/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemResponseDto> findAllByOrderId(Long orderId,
+                                                       Pageable pageable,
+                                                       Long userId) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("There is no order with the id: " + orderId)
+        );
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/OrderServiceImpl.java]: I see that you check it in three  methods let’s  move it to the private method

File: src/main/java/com/bookshop/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        if (!order.getUser().getId().equals(userId)) {
+            throw new EntityNotFoundException(
+                    "There is no order with the id %d in your order history"
+                    .formatted(orderId));
+        }
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/OrderServiceImpl.java]: the same

File: src/main/java/com/bookshop/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        return orderItemRepository.findAllByOrderId(orderId, pageable)
+                .stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto findOrderItemByIdInOrder(Long orderId, Long id, Long userId) {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/OrderServiceImpl.java]:
```suggestion
    public OrderItemResponseDto findOrderItemByIdInOrder(Long orderId, Long orderItemId, Long userId) {
```

File: src/main/java/com/bookshop/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                    .formatted(orderId));
+        }
+        return order.getOrderItems().stream()
+                .filter(i -> i.getId().equals(id))
+                .map(orderItemMapper::toDto)
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "There is no order item with the id: " + id)
+                );
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/OrderServiceImpl.java]: can you return only one object from DB and avoid this filtration?

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: src/main/java/com/bookshop/repository/OrderItemRepository.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.bookshop.repository;
+
+import com.bookshop.model.OrderItem;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface OrderItemRepository extends JpaRepository<OrderItem, Long> {
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookshop/repository/OrderItemRepository.java:10]: `@Repository`

File: src/main/java/com/bookshop/repository/OrderRepository.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package com.bookshop.repository;
+
+import com.bookshop.model.Order;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface OrderRepository extends JpaRepository<Order, Long> {
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookshop/repository/OrderRepository.java:12]: `@Repository`

**Review by Elena-Bruyako — APPROVED** [review]
Good job! Approve, but left two comments

**Pull request merged**